src/Voron/Platform/Win32/WindowsMemoryMapPager.cs

```diff
@@ -131,7 +131,11 @@ public struct SplitValue
 
             NumberOfAllocatedPages = _totalAllocationSize / Constants.Storage.PageSize;
 
-            SetPagerState(CreatePagerState());
+            var pager = CreatePagerState();
+            if (fileAttributes.HasFlag(Win32NativeFileAttributes.Temporary))
+                pager.DiscardDataOnDisk();
```

Comment:
It looks like this is called, and then we even though we _have_ ` if (PlatformDetails.CanDiscardMemory)`, it isn't good enough?

Reply:
No, the problem is when we discard data which is not temporary :)